tests/unit/archive_base_test.py

```diff
@@ -540,3 +543,158 @@ def test_year_filter():
+                    "title": "epacems-1995.zip",
+                    "parts": {
+                        "year_quarter": [
+                            "1995q2",
+                            "1995q3",
+                            "1995q4",
+                        ]
+                    },
+                    "encoding": "utf-8",
+                    "mediatype": "application/zip",
+                    "format": ".zip",
+                    "bytes": 18392692,
+                    "hash": "f4ab7fbaa673fa7592feb6977cc41611",
+                },
+                {
+                    "profile": "data-resource",
+                    "name": "epacems-1996.zip",
+                    "path": "https://zenodo.org/records/10306114/files/epacems-1996.zip",
+                    "remote_url": "https://zenodo.org/records/10306114/files/epacems-1996.zip",
+                    "title": "epacems-1996.zip",
+                    "parts": {
+                        "year_quarter": [
+                            "1996q1",
+                            "1996q2",
+                            "1996q3",
+                            "1996q4",
+                        ]
+                    },
+                    "encoding": "utf-8",
+                    "mediatype": "application/zip",
+                    "format": ".zip",
+                    "bytes": 18921381,
+                    "hash": "fb35601de4e849a6cd123e32f59623f9",
+                },
+                {
+                    "profile": "data-resource",
+                    "name": "epacems-1997.zip",
+                    "path": "https://zenodo.org/records/10306114/files/epacems-1997.zip",
+                    "remote_url": "https://zenodo.org/records/10306114/files/epacems-1997.zip",
+                    "title": "epacems-1997.zip",
+                    "parts": {
+                        "year_quarter": [
+                            "1997q1",
+                            "1997q2",
+                        ]
+                    },
+                    "encoding": "utf-8",
+                    "mediatype": "application/zip",
+                    "format": ".zip",
+                    "bytes": 21734463,
+                    "hash": "e9ece6bb8190e14d44834bf0a068ac5d",
+                },
+            ],
+            "created": "2023-11-30 20:51:43.255388",
+            "version": "1",
+        }
+    )
+)
+fake_new_datapackage_month_success = copy.deepcopy(fake_new_datapackage_quarter_success)
+for resource in fake_new_datapackage_month_success.resources:
+    resource_months_min = pd.to_datetime(min(resource.parts["year_quarter"]))
+    resource_months_max = pd.to_datetime(
+        max(resource.parts["year_quarter"])
+    ) + relativedelta(months=3)
+    dd_idx = pd.date_range(start=resource_months_min, end=resource_months_max, freq="M")
+    resource.parts = {
+        "year_month": [pd.to_datetime(x).strftime("%Y-%m") for x in dd_idx]
+    }
```

Comment:
why is this chunk necessary? because its converting the quarterly datapackage to a monthly datapackage?

Reply:
Yes, this code converts the quarterly test datapackage (the only one fully written out) to a monthly datapackage.